packages/next-auth/src/next/index.ts

```diff
@@ -118,11 +118,14 @@ export async function unstable_getServerSession(
     },
   })
 
-  const { body, cookies } = session
+  const { body, cookies, status } = session
 
   cookies?.forEach((cookie) => setCookie(res, cookie))
 
-  if (body && Object.keys(body).length) return body as Session
+  const error = status && status !== 200
+
+  if (!error && body && Object.keys(body).length) return body as Session
```

Comment:
So I was thinking of another approach. On line 110, we could add `string` to the Generic type:
```ts
  const session = await NextAuthHandler<Session | {} | string>({

```

And check for string here:
```suggestion
  if (body && typeof body !== 'string' && Object.keys(body).length) return body as Session
```

WDYT? This way we don't have to check for status, so the logic will stay the same 😅

Reply:
Awesome, I wasn't 100% sure in my original approach if `status` was a reliable way to check for errors (because technically [ErrorPage sets a 200 status](https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/core/pages/error.tsx#L34) for its default). Checking for `string` is much nicer 👍 Thanks!